addons/payment/controllers/portal.py

```diff
@@ -93,6 +94,10 @@ def payment_status_poll(self):
         tx_to_process = payment_transaction_ids.filtered(lambda x: x.state == 'done' and x.is_processed is False)
         try:
             tx_to_process._post_process_after_done()
+        except psycopg2.OperationalError as e:
+            request.env.cr.rollback()
+            result['success'] = False
+            result['error'] = "pg_error"
```

Comment:
Can you log the error?
```suggestion
            result['error'] = "pg_error"
            _logger.exception("Error while processing transactions")
```

Reply:
I would not use `logger.exception`, the traceback is unnecessary if the situation is expected, taking into account that sql_db will already log the "Bad query" part. You could just log in INFO that the transaction cannot be processed due to another transaction in progress, and will be retried automatically